tribe_client/utils.py

```diff
@@ -121,14 +128,28 @@ def retrieve_public_genesets(options={}, retrieve_all=False):
 
             while meta['next'] is not None:
                 genesets_url = TRIBE_URL + meta['next']
+
+                # Note: Passing in 'options' as the 'params' keyword argument
+                # is no longer necessary since the meta['next'] url already
+                # contains these parameters encoded in the url.
                 tribe_connection = requests.get(genesets_url)
-                result = tribe_connection.json()
-                genesets.extend(result['objects'])
-                meta = result['meta']
+
+                try:
+                    result = tribe_connection.json()
+                    genesets.extend(result['objects'])
+                    meta = result['meta']
+
+                except:
```

Comment:
Since the same exception will be caught on lines 150-153, can we remove this `except` section (lines 142-146) and line 137?

Reply:
Ah, you're totally right @dongbohu, good catch. I'll get rid of that unnecessary try...except